src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs

```diff
@@ -181,6 +171,136 @@ where buildClient.GetArtifactsAsync(build.Project.Id, build.Id, cancellationToke
                     select build).FirstOrDefault();
         }
 
+        private static async Task<Release> CreateReleaseAsync(Build build, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            var releaseClient = ProjectCollection.GetClient<ReleaseHttpClient>();
+            
+            var releaseDefinition = (await releaseClient.GetReleaseDefinitionsAsync(Options.TFSProjectName, ReleaseDefinitionName, cancellationToken: cancellationToken)).FirstOrDefault();
+
+            if (releaseDefinition == null)
+            {
+                Log.Log(NLog.LogLevel.Error, $"Could not find a release definition with name: {ReleaseDefinitionName}");
+                return null;
+            }
+
+            var releaseMetadata = new ReleaseStartMetadata()
+            {
+                DefinitionId = releaseDefinition.Id,
+                Description = $"Automated release for {Options.BranchName} for build {build.BuildNumber}",
+                Reason = ReleaseReason.ContinuousIntegration
+            };
+
+            var artifactMetadata = new ArtifactMetadata
+            {
+                Alias = "Roslyn-Signed", // This is the alias in the Release definition.
```

Comment:
This should pull in the name of whatever build queue we kicked off, right?

Reply:
This is the alias for the release definition specified in the release definition. It isn't tied to the build queue.